src/paradigma/tremor/tremor_analysis.py

```diff
@@ -11,20 +11,61 @@
 
 from paradigma.constants import DataColumns
 from paradigma.config import TremorFeatureExtractionConfig, TremorDetectionConfig, TremorQuantificationConfig
-from paradigma.tremor.feature_extraction import extract_spectral_domain_features
-from paradigma.segmenting import tabulate_windows_legacy
+from paradigma.tremor.feature_extraction import extract_spectral_domain_features, extract_spectral_domain_features
+from paradigma.segmenting import tabulate_windows, tabulate_windows_legacy
 from paradigma.util import get_end_iso8601, write_df_data, read_metadata
 
 
 def extract_tremor_features(df: pd.DataFrame, config: TremorFeatureExtractionConfig) -> pd.DataFrame:
+    """
+    This function groups sequences of timestamps into windows and subsequently extracts 
+    tremor features from windowed gyroscope data.
+
+    Parameters
+    ----------
+    df : pd.DataFrame
+        The input DataFrame containing sensor data, which includes time and gyroscope data. The data should be
+        structured with the necessary columns as specified in the `config`.
+
+    config : TremorFeatureExtractionConfig
+        Configuration object containing parameters for feature extraction, including column names for time, gyroscope data,
+        as well as settings for windowing, and feature computation.
+
+    Returns
+    -------
+    pd.DataFrame
+        A DataFrame containing extracted tremor features and a column corresponding to time.
+    
+    Notes
+    -----
+    - This function groups the data into windows based on timestamps.
+    - The input DataFrame must include columns as specified in the `config` object for proper feature extraction.
+
+    Raises
+    ------
+    ValueError
+        If the input DataFrame does not contain the required columns as specified in the configuration or if any step in the feature extraction fails.
+    """
     # group sequences of timestamps into windows
-    df_windowed = tabulate_windows_legacy(config,df)
+    window_cols = [DataColumns.TIME] + config.gyroscope_cols
+    data_windowed = tabulate_windows(config, df, window_cols)
+
+    idx_time = window_cols.index(DataColumns.TIME)
+    idx_gyro = slice(1, 4)
```

Comment:
You might want to take a look at the changes made for _gait_analysis.py_ in the PR #88. Using `slice` is quite error prone (actually, I ran into problems doing this), therefore you might want to use something like this:

`idx_gyro = extractor.get_slice(config.gyroscope_cols)`